XSConsoleData.py

```diff
@@ -486,9 +487,32 @@ def UpdateFromHostname(self):
             self.ScanHostname(output.split("\n"))
 
     def UpdateFromNTPConf(self):
-        (status, output) = getstatusoutput("/bin/cat /etc/chrony.conf")
-        if status == 0:
-            self.ScanNTPConf(output.split("\n"))
+        if not 'ntp' in self.data:
+            self.data['ntp'] = {}
+
+        if os.path.isfile("/etc/chrony.conf"):
+            with open("/etc/chrony.conf", "r") as confFile:
+                self.ScanNTPConf(confFile.read().splitlines())
+
+        self.data['ntp']['method'] = ""
+        chronyPerm = os.stat("/etc/dhcp/dhclient.d/chrony.sh").st_mode
+        if (
+            chronyPerm & stat.S_IXUSR
+            and chronyPerm & stat.S_IXGRP
+            and chronyPerm & stat.S_IXOTH
+        ):
+            self.data['ntp']['method'] = "DHCP"
+        elif self.data['ntp']['servers']:
+            self.data['ntp']['method'] = "Manual"
+
+            servers = self.data['ntp']['servers']
+            if len(servers) == 4 and all(
```

Comment:
Some coding standards consider the use of magic numbers a defect, I tried to get some references, because I don't understand the meaning of 4 in this context:
> Today, I got feedback from a code review and learned something new. It was about ["magic numbers"](https://en.wikipedia.org/wiki/Magic_number_(programming)), which are unique values with unexplained meaning or multiple occurrences which could (preferably) be replaced with named constants.
> https://dev.to/anthonyjdella/magic-numbers-in-programming-19h0

Initially, skipped adding a review comment about it, but now I recognize that the meaning 4 in this case is not obvious to me. Hence, it may deserve an explanatory comment or (and?) the use of a symbolic constant (if that is sufficient)